storm-client/src/jvm/org/apache/storm/windowing/Window.java

```diff
@@ -47,9 +47,17 @@
     List<T> getExpired();
 
     /**
-     * If processing based on event time, returns the watermark time otherwise the current timestamp.
+     * If processing based on event time, returns the window end time based on watermark otherwise
+     * returns the current timestamp.
```

Comment:
Returning the current timestamp(at which this method is invoked) for non event-timestamp scenarios does not look to be intuitive. Users may assume `endTimeStamp` should return the timestamp at which the window is completed.